### PR TITLE
Chitter

### PR DESCRIPTION
**Pull request by joejknowles**

### CODE REVIEW

**Comment by tansaku** [comment]
:checkered_flag:

File: app/public/css/main.css [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+.cheeps {
+  display: inline-block;
+}
+.cheep {
+  background-color: white;
+  text-align: left;
+  width: 50%;
+  box-shadow: 0px 0px 5px #999;
+  margin: 24%;
```
Comment (joejknowles) [inline app/public/css/main.css]: Use margin: 0 auto;

File: app/server.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+require_relative 'models/send_mail'
+
+class Chitter < Sinatra::Base
+  use Rack::Flash
+  use Rack::MethodOverride
+  include Rack::Utils
+  enable :sessions
+  get '/' do
+    session[:page] = '/'
+    @cheeps = Cheep.all
```
Comment (joejknowles) [inline app/server.rb]: Don't put instance variables in controller

File: app/server.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @user = User.first(id: session[:user_id])
```
Comment (joejknowles) [inline app/server.rb]: write helper method for setting current user

File: app/views/cheeps.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<div id='cheeps'>
+  <% @cheeps.reverse_each do |cheep| %>
+    <div class='cheep'>
+      <img src='/images/default-photo.png'>
+      <div class='author'>
+        <% author = cheep.user.username %>
```
Comment (joejknowles) [inline app/views/cheeps.erb]: author helper method

File: app/views/cheeps.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      <div class='applause'>
+        <%= cheep.applause.length %> applause
+        <% if @user && @user.applause.none? { |ap| ap.cheep_id == cheep.id } && @user != cheep.user%>
+          <form action='./cheeps/applaud' method='post'>
+            <button name='cheepid'value="<%= cheep.id %>">Applaud</button>
+          </form>
+        <% end %>
+      </div>
+      <div class='time'>
+        <%= ((Time.now - cheep.time_stamp) / 60).floor %> minutes ago.
```
Comment (joejknowles) [inline app/views/cheeps.erb]: helper method for time

File: app/controllers/user_management.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
+class Chitter < Sinatra::Base
+  get '/users/new' do
+    erb :'users/new'
+  end
+
+  post '/users/new' do
+    email = params[:email]
+    username = params[:username]
+    password = params[:password]
```
Comment (joejknowles) [inline app/controllers/user_management.rb]: create_user(params)

File: app/controllers/social.rb [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    @user = User.first(id: session[:user_id])
+    erb :'users/profile'
+  end
+
+  post '/users/profiles/follow' do
+    @user = User.first(id: session[:user_id])
+    profile = params[:profile]
+    @profile = User.first(username: profile)
+    @user.follow(@profile)
+    refresh = request.fullpath[0..-7]
```
Comment (joejknowles) [inline app/controllers/social.rb]: This is bad.

File: app/controllers/social.rb [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    end
+    redirect refresh
+  end
+
+  get '/users/profiles/:profile' do
+    @profile = params[:profile]
+    session[:profile] = @profile
+    @profile = User.first(username: @profile)
+    @cheeps = Cheep.all.select do |cheep|
+      cheep.user.username == @profile.username
```
Comment (joejknowles) [inline app/controllers/social.rb]: all these instance variables should be set in helper methods

File: app/controllers/send_cheep.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class Chitter < Sinatra::Base
+  post '/cheeps/new' do
+    cheep = params[:cheep]
+    @user = User.first(id: session[:user_id])
```
Comment (joejknowles) [inline app/controllers/send_cheep.rb]: helper method

**Pull request closed without merging**